internal/provider/resource_time_sleep_test.go

```diff
@@ -192,25 +190,18 @@ func TestAccTimeSleep_CreateDuration(t *testing.T) {
 				ConfigStateChecks: []statecheck.StateCheck{
 					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("create_duration"), knownvalue.StringExact("2ms")),
 					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("id"), knownvalue.NotNull()),
-					captureTimeState2,
+					assertIDUpdated.AddStateValue(resourceName, tfjsonpath.New("id")),
 				},
 			},
 		},
 	})
-
-	// Ensure the id time value is different due to the sleep
-	if captureTimeState1.Value == captureTimeState2.Value {
-		t.Fatal("attribute values are the same")
-	}
```

Comment:
Quick note here since I'm changing the assertion of this test. The original test was actually asserting the values were the same, but I must've missed that in #295. This test has been passing since because it's comparing pointer addresses and not the actual values.